include/wil/safecast.h

```diff
@@ -374,6 +381,43 @@ HRESULT safe_cast_nothrow(const OldT var, NewT* newTResult)
     *newTResult = static_cast<NewT>(var);
     return S_OK;
 }
+
+#if __cpp_if_constexpr
+// This conversion takes a signed integer value and grows it with the upper bits set to zero.  This is
+// useful when the resulting value is cast to a pointer type as it prevents the upper bits from being fill
+// which would adjust the pointed-to address.
+//
+// For example:
+//      wil::safe_zero_extending_cast<ULONG_PTR>(-1)
+// will return 0x00000000`FFFFFFFF on a 64-bit system.
+template <typename NewT, typename OldT, wistd::enable_if_t<details::is_sign_extending_cast_v<NewT, OldT>, int> = 0>
+NewT safe_zero_extending_cast(const OldT var)
+{
+    if constexpr (sizeof(OldT) == sizeof(NewT))

```

Comment:
Can we pull in `type_traits` and say:

```c++
return static_cast<NewT>(static_cast<std::make_unsigned_t<OldT>>(var)));
```

Reply:
My understanding is that WIL avoids STL dependencies outside of a few select headers such as <wil/stl.h>.  I'm unsure of the implications of pulling in a dependency like that.

Reply:
Right, we probably don't want to add _new_ STL dependencies, at least not without validating with a full OS build first. FWIW `wistd::make_unsigned` _does_ already exist

Reply:
Nice, that did the trick without having to pull in any new dependencies.  Thank you for the offline help getting it to compile.